satpy/dataset/metadata.py

```diff
@@ -137,22 +147,71 @@ def _all_arrays_equal(arrays):
     """
     if hasattr(arrays[0], 'compute'):
         return _all_identical(arrays)
-    else:
-        return _pairwise_all(nan_allclose, arrays)
+    return _all_values_equal(arrays)
+
+
+def _all_values_equal(values):
+    return _all_close(values) or _all_equal(values)
```

Comment:
I tried to keep the try...except blocks in separate methods and to avoid nesting them. So this is what I came up with. The problem: `_all_close` and `_all_equal` return `False` if the values are not equal *or* the types don't match. So if the values are comparable but not close, they are compared twice here. Do you have an idea how to avoid this?

Reply:
In what situation would you have a `_all_close` fail because of the type but `_all_equal` work?

Reply:
With datetime arrays for example

Reply:
I think you need `_all_close` to raise a `TypeError` or something then, and then
```python
try:
    return _all_close(values)
except:
    return _all_equal(values)
```

Reply:
Perfect, thanks!